### PR TITLE
[TINY] Fix to #19606 - Query: add support for convert to/from bool value using Convert.To* methods

### LINKED ISSUE #19606: Query: add support for convert to/from bool value using Convert.To* methods
We dont support bool currently. Could be useful in some cases, e.g. https://github.com/dotnet/efcore/issues/19553

### PR DESCRIPTION
**Pull request by maumar**
Fixes #19606

### CODE REVIEW

File: test/EFCore.SqlServer.FunctionalTests/Query/NorthwindFunctionsQuerySqlServerTest.cs [hunk 1]
```diff
@@ -943,6 +943,10 @@ public override async Task Convert_ToByte(bool async)
             AssertSql(
                 @"SELECT [o].[OrderID], [o].[CustomerID], [o].[EmployeeID], [o].[OrderDate]
 FROM [Orders] AS [o]
+WHERE ([o].[CustomerID] = N'ALFKI') AND (CONVERT(tinyint, CONVERT(bit, [o].[OrderID] % 1)) >= CAST(0 AS tinyint))",
```
Comment (smitpatel) [inline test/EFCore.SqlServer.FunctionalTests/Query/NorthwindFunctionsQuerySqlServerTest.cs:946]: Assert SQL for Convert_ToBoolean test?

**Comment by maumar** [comment]
@smitpatel updated

File: test/EFCore.SqlServer.FunctionalTests/Query/QueryBugsTest.cs [hunk 2]
```diff
@@ -7277,6 +7277,88 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
 
         #endregion
 
+        [ConditionalFact]
+        public void Fubarson()
```
Comment (smitpatel) [inline test/EFCore.SqlServer.FunctionalTests/Query/QueryBugsTest.cs:7281]: What is this? :trollface:

Comment (maumar) [reply test/EFCore.SqlServer.FunctionalTests/Query/QueryBugsTest.cs:7281]: VS glitched again and didnt save changes >.<

**Pull request merged**